### PR TITLE
Add i18n

### PR DESCRIPTION
**Pull request by CharlonTank**
Added I18n for

- [x] English
- [x] French
- [x] Spanish

Added a select to choose language.

`navigator.language` is used to get browser's set language then local storage `language` variable.

### CODE REVIEW

**Comment by MartinSStewart** [comment]
I had a quick look at the code and it looks like what I had in mind 👍 

Good work so far!

**Comment by MartinSStewart** [comment]
Awesome! I'll take a closer look at this over the weekend.

**Comment by MartinSStewart** [comment]
> In a second time, what would be nice is to match the language based on the timezone.

It looks like France is in the CET timezone. I live in Sweden which is also CET but I don't know any French. It seems likely that a lot of other people will be in CET but not be French speakers? Or maybe I'm misunderstanding something 😅

**Comment by CharlonTank** [comment]
> > In a second time, what would be nice is to match the language based on the timezone.
> 
> It looks like France is in the CET timezone. I live in Sweden which is also CET but I don't know any French. It seems likely that a lot of other people will be in CET but not be French speakers? Or maybe I'm misunderstanding something 😅

Yeah, I need to understand how to do i18n properly, I guess there is a way to know the language of the user based on some browser data, do you know about that?

**Comment by CharlonTank** [comment]
I now use the browser language (`navigator.language`) in addition to the local storage value.

File: src/Frontend.elm
Comment (MartinSStewart) [inline src/Frontend.elm:31]: I'm a bit undecided whether to alias `Element.*` modules. For now though, so that it's consistent with existing code, can you change back all the instances of `import Element.* as *` to just being `import Element.*`?

File: src/Frontend.elm
Comment (MartinSStewart) [inline src/Frontend.elm:1634]: This repo already breaks this rule but avoid using `<|`. Stick to using parenthesis or `|>` (I should add https://package.elm-lang.org/packages/SiriusStarr/elm-review-pipeline-styles/latest/ to the elm-review config). My reasoning for this is that `<|` doesn't look good if there's line breaks (especially if there's multiple `<|`) and it's entirely redundant so sticking with `|>` and parenthesis reduces the amount of decisions that need to be made.

**Review by MartinSStewart — CHANGES_REQUESTED** [review]
I've made some comments on code style fixes. It looks like I can make changes to this PR as well so I'm going to tweak some stuff such as the look of the language dropdown if you don't mind.

If you're busy I can go ahead and handle the style fixes too but I figured I'd give an explanation on why things are the way they are.

Lastly, now that I've had a closer look at this PR, I'm really impressed by the amount of work you've put in. I really appreciate the commitment!

File: src/Frontend.elm
Comment (CharlonTank) [reply src/Frontend.elm:31]: Sure, no problem! As I said in Discord, I was kinda, let's go and if you don't like my alias normalisation, I could just go back to where it was before. So sure :)

File: src/Frontend.elm
Comment (CharlonTank) [reply src/Frontend.elm:30]: I'll keep the `Element.*`

File: src/Frontend.elm
Comment (CharlonTank) [reply src/Frontend.elm:1634]: Alright 👍

**Comment by CharlonTank** [comment]
> I've made some comments on code style fixes. It looks like I can make changes to this PR as well so I'm going to tweak some stuff such as the look of the language dropdown if you don't mind.
> 
> If you're busy I can go ahead and handle the style fixes too but I figured I'd give an explanation on why things are the way they are.
> 
> Lastly, now that I've had a closer look at this PR, I'm really impressed by the amount of work you've put in. I really appreciate the commitment!

You also gave me the motivation to improve this project by a little so I'm really happy :)

I'm gonna put `Element.*` back and use `|>`  everywhere I can. https://package.elm-lang.org/packages/SiriusStarr/elm-review-pipeline-styles/latest/ seems nice!

Regarding my use of `<|` over `|>`, it's because I believe that the function is more important for the developer to see immediately rather than the arguments of the function. For instance, I find `htmlAttribute` to be a distraction, but `inFront` is vital since it's not just a simple Attribute msg.

Following your principle, I have to user parenthesis for `inFront` like you did already.

Thanks again for your encouragement and for your feedback. It means a lot to me!

**Comment by MartinSStewart** [comment]
Before I merge this I'd like to add visual regression testing. Unfortunately I have another project I'm working on right now and I'm going to need to put this PR on hold for 2 weeks. Sorry about the delay!

**Comment by CharlonTank** [comment]
> Before I merge this I'd like to add visual regression testing. Unfortunately I have another project I'm working on right now and I'm going to need to put this PR on hold for 2 weeks. Sorry about the delay!

No problem! Nice that you're adding VRT anyway! I'm curious about that too!

**Comment by MartinSStewart** [comment]
Okay, finally merged! Thanks again for the help @CharlonTank! Here's the visual regression snapshots if you're curious https://percy.io/7bbb05a0/meetdown. The setup for it is here https://github.com/MartinSStewart/meetdown/tree/master/snapshot-runner and the snapshots themselves are mostly in this test https://github.com/MartinSStewart/meetdown/blob/eed9a65c84e6ef5bac6296bc95eb4a7b59caa702/src/Tests.elm#L1425

**Pull request merged**
